v2/azure-arm.yaml

```diff
@@ -1132,6 +1132,10 @@ objectModelConfiguration:
         $export: true
         $supportedFrom: v2.4.0
         $defaultAzureName: false
+        $operatorSpecProperties:
+          - name: UUIDGeneration
```

Comment:
Is this the best name for the property? 

I used `NamingConvention` in my demo of the OperatorSpec implementation; maybe there's something better than either?

Reply:
Also, don't we need to specify this on both versions of `RoleAssignment`, and both versions of `RoleDefinition`?

Reply:
Bit unsure, should we specify on both versions?

Reply:
Do you have a suggestion for naming? or something from the list below you prefer? 
- `GenerationType`
- `NamingConvention`
- `AzureNameGeneration`

Reply:
I like `NamingConvention` (obviously!) - also like `AzureNameGeneration` too.

Reply:
Naming anything is hard. Naming about naming, doubly so! 😁